packages/components/src/internal/components/navigation/NavigationBar.spec.tsx

```diff
@@ -65,100 +82,154 @@ describe('NavigationBar', () => {
+            getDefaultAppContext(),
+            getDefaultServerContext()
         );
+        await waitForLifecycle(component);
         validate(component, { FindAndSearchDropdown: 0, SearchBox: 0 });
         component.unmount();
     });
 
-    test('with notifications no user', () => {
+    test('with notifications no user', async () => {
         const component = mountWithAppServerContext(
-            <NavigationBar model={productMenuModel} notificationsConfig={notificationsConfig} />
+            <NavigationBar notificationsConfig={notificationsConfig} />,
+            getDefaultAppContext(),
+            getDefaultServerContext()
         );
+        await waitForLifecycle(component);
         validate(component, { ServerNotifications: 0 });
         component.unmount();
     });
 
-    test('with notifications, guest user', () => {
+    test('with notifications, guest user', async () => {
         const component = mountWithAppServerContext(
-            <NavigationBar model={productMenuModel} user={TEST_USER_GUEST} notificationsConfig={notificationsConfig} />
+            <NavigationBar user={TEST_USER_GUEST} notificationsConfig={notificationsConfig} />,
+            getDefaultAppContext(),
+            getDefaultServerContext()
         );
+        await waitForLifecycle(component);
         validate(component, { UserMenu: 1, ServerNotifications: 0 });
         component.unmount();
     });
 
-    test('with notifications, non-guest user', () => {
+    test('with notifications, non-guest user', async () => {
         const component = mountWithAppServerContext(
-            <NavigationBar model={productMenuModel} user={TEST_USER_READER} notificationsConfig={notificationsConfig} />
+            <NavigationBar user={TEST_USER_READER} notificationsConfig={notificationsConfig} />,
+            getDefaultAppContext(),
+            getDefaultServerContext()
         );
+        await waitForLifecycle(component);
         validate(component, { UserMenu: 1, ServerNotifications: 1 });
         component.unmount();
     });
 
-    test('show ProductNavigation for hasPremiumModule, non-admin', () => {
-        const component = mountWithAppServerContext(
-            <NavigationBar model={productMenuModel} user={TEST_USER_READER} />,
-            undefined,
-            { moduleContext: { api: { moduleNames: ['premium'], applicationMenuDisplayMode: 'ALWAYS' } } }
-        );
+    test('show ProductNavigation for hasPremiumModule, non-admin', async () => {
+        const component = mountWithAppServerContext(<NavigationBar user={TEST_USER_READER} />, getDefaultAppContext(), {
+            container: TEST_PROJECT_CONTAINER,
+            moduleContext: {
+                api: {
+                    moduleNames: ['samplemanagement', 'premium'],
+                    applicationMenuDisplayMode: 'ALWAYS',
+                },
+                samplemanagement: { productId: SAMPLE_MANAGER_APP_PROPERTIES.productId },
+            },
```

Comment:
You could possibly use the `TEST_LKS_STARTER_MODULE_CONTEXT` constant here and augment with the `applicationMenuDisplayMode` value.

Reply:
Good suggestion. Done.